### PR TITLE
Add a Pull Request template

### PR DESCRIPTION
**Pull request by cmaloney**

### CODE REVIEW

File: PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+High level description including:
```
Comment (orsenthil) [inline PULL_REQUEST_TEMPLATE.md]: **High Level Description**

- Capitalize
- Format Bold

File: PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+ - What this change is enabling / fixing.
```
Comment (orsenthil) [inline PULL_REQUEST_TEMPLATE.md]: `What does this change enable?`

File: PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+ - High level description of how things changed
+
+# Issues
+
+[DCOS-<number>](https://mesosphere.atlassian.net/browse/DCOS-<number>)
```
Comment (orsenthil) [inline PULL_REQUEST_TEMPLATE.md]: Is it mesosphere.atlassian or dcos.atlassian ?

File: PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+...
+
+# Checklist
+
+ - [ ] Included a test which will fail if code is reverted but test is not
```
Comment (orsenthil) [inline PULL_REQUEST_TEMPLATE.md]: `Included a test case for this change`

File: PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+ - [ ] Follows relevenat code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)
```
Comment (orsenthil) [inline PULL_REQUEST_TEMPLATE.md]: typo: relevant
`Followed`

File: PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (cmaloney) [inline PULL_REQUEST_TEMPLATE.md]: This isn't a title or intended to be one. It's a "placeholder text you should replace with your own text in this formatting"

File: PULL_REQUEST_TEMPLATE.md (hunk 4 above)
Comment (cmaloney) [inline PULL_REQUEST_TEMPLATE.md]: The revert part is critical to what's being asked for. People will always say they "tested it in their component", the goal is to get across we want to test the _integration_ which is distinct from testing that the feature works at all (which unit and integration tests inside their component cover)

File: PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+# Issues
+
+[DCOS-<number>](https://dcosjira.atlassian.net/browse/DCOS-<number>)
+...
+
+# Checklist
+
+ - [ ] Included a test which will fail if code is reverted but test is not
+ - [ ] Followed relevenat code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)
```
Comment (orsenthil) [inline PULL_REQUEST_TEMPLATE.md]: typo: relevenat should be relevant

**Pull request merged**